Marlin/Marlin_main.cpp

```diff
@@ -9496,39 +9496,6 @@ inline void gcode_M502() {
 
 #endif // ADVANCED_PAUSE_FEATURE
 
-#if ENABLED(MK2_MULTIPLEXER)
-
-  inline void select_multiplexed_stepper(const uint8_t e) {
-    stepper.synchronize();
-    disable_e_steppers();
-    WRITE(E_MUX0_PIN, TEST(e, 0) ? HIGH : LOW);
-    WRITE(E_MUX1_PIN, TEST(e, 1) ? HIGH : LOW);
-    WRITE(E_MUX2_PIN, TEST(e, 2) ? HIGH : LOW);
-    safe_delay(100);
-  }
-
-  /**
-   * M702: Unload all extruders
-   */
-  inline void gcode_M702() {
-    for (uint8_t s = 0; s < E_STEPPERS; s++) {
-      select_multiplexed_stepper(e);
-      // TODO: standard unload filament function
-      // MK2 firmware behavior:
-      //  - Make sure temperature is high enough
-      //  - Raise Z to at least 15 to make room
-      //  - Extrude 1cm of filament in 1 second
-      //  - Under 230C quickly purge ~12mm, over 230C purge ~10mm
-      //  - Change E max feedrate to 80, eject the filament from the tube. Sync.
-      //  - Restore E max feedrate to 50
-    }
-    // Go back to the last active extruder
-    select_multiplexed_stepper(active_extruder);
-    disable_e_steppers();
-  }
-
-#endif // MK2_MULTIPLEXER
```

Comment:
I'm just not a big fan of having to manually command the multiplexer and synchronize to the steppers (thereby stopping the command processing pipeline) in every location that needs to change the multiplexer. All of that duplication will ultimately result in a bug when someone forgets to make one of those calls, which could potentially be difficult to track down because the steppers will end up switching prematurely. 

Speaking of cleaning up the Stepper isr, I noticed that the stepper direction signals are set at the beginning of each block. I also noticed that each pulse pin is set sequentially, which results in a DIO hardware register write for each stepper being pulsed. I wonder if the bits could be written into a CPU register and then all written out in one shot. Seems like it could cut down on some cycles.